aya-obj/src/btf/btf.rs

```diff
@@ -821,6 +899,139 @@ impl Default for Btf {
 }
 
 impl Object {
+    fn patch_extern_data_internal(
+        &mut self,
+        externs: &HashMap<String, Vec<u8>>,
+    ) -> Result<Option<(SectionIndex, Vec<u8>)>, BtfError> {
+        if let Some(obj_btf) = &mut self.btf {
+            if obj_btf.is_empty() {
+                return Ok(None);
+            }
+
+            // Build a single-pass index of all VARs contained in DATASEC types:
+            // var_name -> (datasec_name, var)
+            let mut datasec_var_index: HashMap<String, (String, Var)> = HashMap::new();
+            for t in &obj_btf.types.types {
+                let BtfType::DataSec(d) = t else {
+                    continue;
+                };
+                let sec_name = obj_btf.string_at(d.name_offset)?.into_owned();
+                for entry in &d.entries {
+                    let BtfType::Var(var) = obj_btf.types.type_by_id(entry.btf_type)? else {
+                        continue;
+                    };
+                    let var_name = obj_btf.string_at(var.name_offset)?.into_owned();
+                    datasec_var_index.insert(var_name, (sec_name.clone(), var.clone()));
+                }
+            }
+
+            let kconfig_map_index = self.maps.len();
+
+            let symbols = self
+                .symbol_table
+                .iter_mut()
+                .filter(|(_, s)| s.name.is_some() && s.section_index.is_none() && s.is_external)
+                .map(|(_, s)| (s.name.as_ref().unwrap().clone(), s));
+
+            let mut section_data = Vec::<u8>::new();
+            let mut offset = 0u64;
+            let mut has_extern_data = false;
+
+            for (name, symbol) in symbols {
+                let (datasec_name, var) = match datasec_var_index.get(&name) {
+                    Some((sec_name, var)) => {
+                        if var.linkage == VarLinkage::Extern {
+                            (sec_name.as_str(), var)
+                        } else {
+                            return Err(BtfError::InvalidExternalSymbol { symbol_name: name });
+                        }
+                    }
+                    None => {
+                        return Err(BtfError::ExternalSymbolNotFound { symbol_name: name });
+                    }
+                };
+
+                if datasec_name == ".kconfig" {
+                    has_extern_data = true;
+
+                    let type_size = obj_btf.type_size(var.btf_type)?;
+                    let type_align = obj_btf.type_align(var.btf_type)? as u64;
+
+                    let mut external_value_opt = externs.get(&name);
+                    let empty_data = vec![0; type_size];
+
+                    if external_value_opt.is_none() && symbol.is_weak {
```

Comment:
explain this? A symbol can be weak but still have a definition?

Reply:
answering myself: weak config symbols get a 0 default value.

This is needed to handle missing CONFIG_* keys.

The code should say so, and we should make sure we do this kconfig biz only for stuff actually in the `.kconfig` section